src/software/thunderscope/robot_diagnostics/handheld_device_manager.py

```diff
@@ -0,0 +1,341 @@
+
+        self.lock = RLock()
+
+        self.logger = logger
+        self.proto_unix_io = proto_unix_io
+
+        self.enabled = False
+
+        self.handheld_device: Optional[InputDevice] = None
+        self.handheld_device_config: Optional[DeviceConfig] = None
+
+        self.__initialize_default_motor_control_values()
+        self.__initialize_default_power_control_values()
+
+        # The following fields are here to temporarily persist device input.
+        # They are read once certain buttons are pressed on the handheld device,
+        # and the values are set for the control primitives above.
+        self.kick_power: float = DiagnosticsConstants.MIN_KICK_POWER
+        self.chip_distance: float = DiagnosticsConstants.MIN_CHIP_POWER
+        self.dribbler_speed: int = 0
+        self.dribbler_running: bool = False
+
+        self.constants = tbots_cpp.create2021RobotConstants()
+
+        self.event_loop_thread = Thread(target=self.__event_loop, daemon=True)
+        self.event_loop_thread.start()
+
+    def __initialize_default_motor_control_values(self) -> None:
+        """Set all required fields in the motor control proto to their
+        default/minimum values
+        """
+        self.motor_control = MotorControl()
+        self.motor_control.direct_velocity_control.velocity.x_component_meters = 0.0
+        self.motor_control.direct_velocity_control.velocity.y_component_meters = 0.0
+        self.motor_control.direct_velocity_control.angular_velocity.radians_per_second = 0.0
+        self.motor_control.dribbler_speed_rpm = 0
+
+    def __initialize_default_power_control_values(self) -> None:
+        """Set all required fields in the power control proto to their
+        default/minimum values
+        """
+        self.power_control = PowerControl()
+        self.power_control.geneva_slot = 3
+
+    def set_enabled(self, enabled) -> None:
+        """Enable or disable the HandheldDeviceManager from processing input events.
+
+        :param enabled: whether to enable (True) or disable (False) the
+                        HandheldDeviceManager from processing input events
+        """
+        with self.lock:
+            self.enabled = enabled
+
+    def reinitialize_handheld_device(self) -> None:
+        """Attempt to reinitialize a connection to a handheld device."""
+        self.__clear_handheld_device()
+        self.__initialize_handheld_device()
+
+    def __initialize_handheld_device(self) -> None:
+        """Attempt to initialize a connection to a handheld device.
+        The first handheld device that is recognized as valid will be used.
+        Valid handheld devices are any devices whose name has a matching DeviceConfig
+        in DiagnosticsConstants.HANDHELD_DEVICE_NAME_CONFIG_MAP.
+        """
+        device_config_map = DiagnosticsConstants.HANDHELD_DEVICE_NAME_CONFIG_MAP
+
+        for device in list_devices():
+            handheld_device = InputDevice(device)
+            if (
```

Comment:
During the test with Arun, we encountered an issue where `handheld_device.name` was not present in `device_config_map`. This led to some confusion among users regarding the status of the controller.

To improve clarity, could we add a debug log message indicating when a device is found? This would help users understand the situation better.

Reply:
alternatively, we could strip whitespace and match lowercase controller names to stop these kinds of issues